docs/arch/Proposals/adr-2504-bulkupdate.md

```diff
@@ -0,0 +1,155 @@
+# ADR Introduce `$bulk-update` FHIR Operation
+*Labels*: [Specification](https://github.com/microsoft/fhir-server/labels/Specification) | [API](https://github.com/microsoft/fhir-server/labels/Area-API) | [Asynchronous-Jobs](https://github.com/microsoft/fhir-server/labels/Area-Asynchronous-Jobs)
+
+---
+
+## Context
+Our platform already supports the FHIR `$import` feature for incremental loads. Customers need a **large‑scale partial‑update capability** that:
+- re‑uses **FHIR Patch** semantics (encoded as a `Parameters` resource — **not** JSON Patch)
+- leverages the [FHIR Asynchronous pattern](https://hl7.org/fhir/async.html)
+- lets them choose resources with familiar expressive search, including `_include` / `_revinclude`
+- operates at **system** scope (`/$bulk-update`) or **type** scope (`/[ResourceType]/$bulk-update`).
+- enables updating specific field values per resource type based on the provided `resource type field mapping.`
+- updates `metadata` properties, such as `version` and `last updated date`, whenever a resource is modified.
+- provides a query to retrieve the list of resources that will be updated.
+- returns a response detailing the `resource count succeeded and resource count failed to update per resource type, along with the failure reasons.`
+- Logs an audit entry when the bulk update is triggered, including the `list of resource IDs successfully updated` and `any errors with the list of resource IDs that failed to update.`
```

Comment:
The audit log isn't made when the job is triggered, but while it is running. This lets it log the progress it has made.